cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -19,7 +21,10 @@ import cromwell.api.CromwellClient._
 
 import scala.util.{Failure, Success, Try}
 
-class CromwellClient(val cromwellUrl: URL, val apiVersion: String)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) {
+class CromwellClient(val cromwellUrl: URL, val apiVersion: String, val credentials: Option[HttpCredentials]=None)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) {
+
+  lazy val authHeader = credentials.map { Authorization(_) }
+  lazy val commonHeaders = authHeader.toList
```

Comment:
Why not just `toList` it up above? This is only being used in the one place down below, not sure I understand the multiple layers here.

Reply:
I could be persuaded that splitting it up like this makes things more readable/understandable and the compiler will probably inline it anyway if it gets called a lot. So, I'd be happy either way.

Reply:
Don't completely disagree w/ that but my main question was having two basically the same variables at the top level which ultimately are only used in the one function. And really the question part of my main question was if there was a reason for that setup :)

Reply:
Yeah, I thought for clarity and a little future-proofing that `executeRequest` should add all common headers, and creating the List of them should be separate from creating each one of them from possibly optional pieces.

Reply:
I was almost going to suggest inlining the whole thing in the one spot so you can see where my biases lie :) Not a big deal, carry on. Will merge this, thanks @delocalizer